src/components/Table/GridTable.test.tsx

```diff
@@ -2421,6 +2421,138 @@ describe("GridTable", () => {
   });
 
+  it("can render with rows with initSelected defined include the children rows", async () => {
+    type HeaderRow = { kind: "header"; id: string; data: undefined };
+    type ParentRow = { kind: "parent"; id: string; data: { name: string } };
+    type ChildRow = { kind: "child"; id: string; data: { name: string } };
+    type GrandChildRow = { kind: "grandChild"; id: string; data: { name: string } };
+    type Row = HeaderRow | ParentRow | ChildRow | GrandChildRow;
+
+    const selectCol = selectColumn<Row>();
+    const nameCol: GridColumn<Row> = {
+      header: "Name",
+      parent: ({ name }) => name,
+      child: ({ name }) => name,
+      grandChild: ({ name }) => name,
+      mw: "160px",
+    };
+
+    // Given rows initially set as selected
+    const rows: GridDataRow<Row>[] = [
+      simpleHeader,
+      {
+        kind: "parent",
+        id: "1",
+        data: { name: "Howard Stark" },
+        initSelected: true,
+        inferSelectedState: false,
+        children: [
+          {
+            kind: "child" as const,
+            id: "2",
+            data: {
+              name: "Tony Stark",
+            },
+            initSelected: false,
+            children: [
+              {
+                kind: "grandChild" as const,
+                id: "5",
+                data: {
+                  name: "Morgan Stark",
+                },
+                initSelected: true,
+              },
+            ],
+          },
+        ],
+      },
+      {
+        kind: "parent",
+        id: "3",
+        data: { name: "Odin" },
+        initSelected: false,
+        inferSelectedState: false,
+        children: [
+          {
+            kind: "child" as const,
+            id: "4",
+            data: {
+              name: "Thor",
+            },
+            initSelected: true,
+          },
+          {
+            kind: "child" as const,
+            id: "6",
+            data: {
+              name: "Hela",
+            },
+            initSelected: true,
+          },
+          {
+            kind: "child" as const,
+            id: "7",
+            data: {
+              name: "Loki",
+            },
+            initSelected: true,
+          },
+        ],
+      },
+    ] as GridDataRow<Row>[];
+
+    const api: MutableRefObject<GridTableApi<Row> | undefined> = { current: undefined };
+
+    function Test() {
+      const _api = useGridTableApi<Row>();
+      api.current = _api;
+      return <GridTable<Row> api={_api} columns={[selectCol, nameCol]} rows={rows} />;
+    }
+
+    // When rendering the table
+    const r = await render(<Test />);
+
+    expect(tableSnapshot(r)).toMatchInlineSnapshot(`
+      "
+      | on | Name         |
+      | -- | ------------ |
+      | on | Howard Stark |
+      | on | Tony Stark   |
+      | on | Morgan Stark |
+      | on | Odin         |
+      | on | Thor         |
+      | on | Hela         |
+      | on | Loki         |
```

Comment:
Do you really need 7 rows for this test case? If so, can you put an `// And ..what makes this row different...` before each of the `const rows`. As-is there are ~130 LOC that I can't really tell what is 100% necessary and what is extra rows just creating cruft.